src/log_prob.jl

```diff
@@ -0,0 +1,438 @@
+        Categorical(pp.k[:,2]),
+        [MvNormal(pp.c_mean[:,k,2], pp.c_cov[:,:,k,2]) for k in 1:2],
+        LogNormal(0, 10),
+        Beta(1, 1),
+        Beta(1, 1))
+
+    return Prior(Bernoulli(.5), star_prior, gal_prior)
+end
+
+
+"""
+Return the log prior probability of a (brightness,color) pair based on
+prior probability objects constructed by `construct_prior()`
+"""
+function color_logprior(brightness::Float64,
+                        colors::Vector{Float64},
+                        prior::Prior,
+                        is_star::Bool)
+    subprior      = is_star ? prior.star : prior.galaxy
+    ll_brightness = logpdf(subprior.brightness, exp(brightness))
+    ll_component  = [logpdf(subprior.colors[k], colors) for k in 1:2]
+    ll_color      = logsumexp(ll_component + log(subprior.color_component.p))
+    return ll_brightness + ll_color
+end
+
+
+function shape_logprior(gal_shape::Vector{Float64}, prior::Prior)
+    # position and gal_angle have uniform priors--we ignore them
+    gdev, gaxis, gangle, gscale = gal_shape
+    ll_shape = 0.
+    ll_shape += logpdf(prior.galaxy.gal_scale, gscale)
+    ll_shape += logpdf(prior.galaxy.gal_ab, gaxis)
+    ll_shape += logpdf(prior.galaxy.gal_fracdev, gdev)
+    return ll_shape
+end
+
+
+#################################################################
+# transformation of fluxes = (u, g, r, i, z) in nanomaggies to  #
+# (bright,color) pairs                                          #
+#################################################################
+
+
+"""
+Translate between fluxes (nmgy rates) and colors (on which
+LogNormal priors are placed)
+"""
+function fluxes_to_colors(fluxes::Vector{Float64})
+    lnr    = log(fluxes[3])
+    colors = Array(Float64, 4)
+    colors[1] = log(fluxes[2]) - log(fluxes[1])
+    colors[2] = log(fluxes[3]) - log(fluxes[2])
+    colors[3] = log(fluxes[4]) - log(fluxes[3])
+    colors[4] = log(fluxes[5]) - log(fluxes[4])
+    return lnr, colors
+end
+
+
+"""
+Translate from the (brightness, color) parameterization to nmgy fluxes
+"""
+function colors_to_fluxes(brightness::Float64, colors::Vector{Float64})
+    # build up log fluxes
+    ret    = Array(Float64, 5)
+    lnr    = brightness
+    ret[3] = lnr     # r flux
+    ret[4] = lnr + colors[3]        # ln(i/r) = c3 => lni = lnr - c3
+    ret[5] = ret[4] + colors[4]     # ln(z/i) = c4 => lnz = lni - c4
+    ret[2] = -colors[2] + lnr       # ln(r/g) = c2 => lng = c2 + lnr
+    ret[1] = -colors[1] + ret[2]    # ln(g/u) = c1 => lnu = c1 + lng
+    return exp(ret)
+end
+
+
+#######################################################################
+# galaxy shape transformation (unconstrained => constrained and back) #
+#######################################################################
+
+
+function constrain_gal_shape(unc_gal_shape::Vector{Float64})
+    gdev, gaxis, gangle, gscale = unc_gal_shape
+    constr_shape    = Array(Float64, 4)
+    constr_shape[1] = clamp(sigmoid(gdev), EPS, 1-EPS)
```

Comment:
This is fine, but just in case you want to avoid the nonlinearity of ```clamp```, you could use ```unbox_parameter```.